Adding skflow to tensorflow/contrib

ilblackdragon opened the pull request.

terrytangyuan committed:
* Fixed dask data extractors

terrytangyuan committed:
* dask extractors are consistent with other io extractors

ilblackdragon committed:
Merge pull request #73 from terrytangyuan/patch-1

Temporarily omit some io stuff from coverage

terrytangyuan committed:
Merge pull request #76 from terrytangyuan/dask

Fixed dask_io.py

terrytangyuan committed:
* Fix lint issues

nicolasfauchereau committed:
Modified the boston.py example so that the standard scaler is fitted over the training set, not the whole dataset, and then applied to the test set

terrytangyuan committed:
Merge pull request #78 from nicolasfauchereau/master

scaler fitted to the training set only, then applied to test set

ilblackdragon committed:
Updated README.md - change google->tensorflow

terrytangyuan committed:
Merge pull request #77 from tensorflow/dask

Fix lint issues

ilblackdragon committed:
Reverse for vocab processor and handling incorrect utf for byte processor reverse

ilblackdragon committed:
Reverse class id to class name for CategoricalVocabulary

ilblackdragon committed:
fix linters

ilblackdragon committed:
Continue working on neural translation model - added real dataset, train/test split, streaming and processing with byte processor and output results

terrytangyuan committed:
+ Basic test for dask io

ilblackdragon committed:
Fixes #79: Added custom dropout op, that adds probability tensor to the collection DROPOUTS and zeros it on prediction

terrytangyuan committed:
+ Tests for dask.dataframe and dask.series

terrytangyuan committed:
+ Minor fixes in dask_io

terrytangyuan committed:
- Include dask_io.py in coverage test

terrytangyuan committed:
Merge pull request #82 from tensorflow/dask

dask_io test coverage

ilblackdragon committed:
Test for dropout probability be a tensor

ilblackdragon committed:
Merge pull request #81 from tensorflow/dropout-fix

Fixes #79: Zeros dropouts on prediction

ilblackdragon committed:
Added Mnist example that showcases skflow for tutorial from https://www.tensorflow.org/versions/master/tutorials/mnist/pros/index.html#deep-mnist-for-experts

makseq committed:
Fixes.

makseq committed:
Fixes for pylint.

terrytangyuan committed:
* Get around with bug in dask

terrytangyuan committed:
* Fixed DaskDataFeeder

terrytangyuan committed:
+ Test for DaskDataFeeder

terrytangyuan committed:
* Working dask_data_feeder test

terrytangyuan committed:
* Fixed lint

terrytangyuan committed:
* Finished test for dask_data_feeder

terrytangyuan committed:
+ Turn on coverage test for data feeder

terrytangyuan committed:
Merge branch 'dask'

terrytangyuan committed:
+ DaskDataFeeder support multi-dim

terrytangyuan committed:
- minor changes in test

terrytangyuan committed:
* minor refactoring

terrytangyuan committed:
+ Integrate DaskDataFeeder with estimators

terrytangyuan committed:
* Fixed column name issue and sample_fraction

terrytangyuan committed:
+ Fixed repeated colnames when concat two dfs with default numeric colnames

terrytangyuan committed:
* Fixed lint

terrytangyuan committed:
+ Fixed inconsistent sized batch and added note

terrytangyuan committed:
+ one-hot encode per class for dask feeder and re-wrote tests

terrytangyuan committed:
+ Check for pd.Series inferred from sampled out and fit is successful

terrytangyuan committed:
* DaskDataFeeder works with iris with same accuracy

terrytangyuan committed:
* prediction on each partition

terrytangyuan committed:
+ Example of out-of-core dataframe classification

terrytangyuan committed:
* Modification of out-of-core example

terrytangyuan committed:
* Minor modification in out-of-core example

okoriko committed:
Fixed the usage of metrics.accuracy_score().

ilblackdragon committed:
ignore install six for tensorflow to get 2.7 work

makseq committed:
Tests passed.

makseq committed:
Unittests changed.

makseq committed:
pylint check fixed.

terrytangyuan committed:
Merge pull request #86 from Phybbit/master

Consistent usage of metrics.accuracy_score with the documentation

terrytangyuan committed:
+ rnn_estimator

terrytangyuan committed:
+ RNNClassifier and RNNRegressor

terrytangyuan committed:
+ input_op_fn for rnn_estimator

terrytangyuan committed:
* minor changes

terrytangyuan committed:
- revert last commit

ilblackdragon committed:
Fixes #84: Reshaping the labels for regression to compute correctly cost for batch_size>1. Also started working on class_weights.

ilblackdragon committed:
removing debugging code

ilblackdragon committed:
Bumped up the default number of steps to get more training if user doesn't provide steps variable.

ilblackdragon committed:
Updated tests to use tf.test.TestCase instead of googletest. Added test_regression for recreating weights

terrytangyuan committed:
* Fixed all lint/nosetests errors

ilblackdragon committed:
Added weights and biases properties to DNN estimators

ilblackdragon committed:
Updated readme - added more about types of models and bumped min requirement for TF to 0.6

ilblackdragon committed:
More bolding and formatting in README

terrytangyuan committed:
+ Example of using built-in rnn model

terrytangyuan committed:
+ Tests for rnn models

terrytangyuan committed:
* Fixed failed test due to change of default for steps in linreg since it overfits

terrytangyuan committed:
+ More tmp coverage for RNN model

terrytangyuan committed:
Merge branch 'rnn'

terrytangyuan committed:
+ Default values for rnn models

terrytangyuan committed:
+ weights and bias for rnn

ilblackdragon committed:
Merge branch 'datafeeder_sampling' of https://github.com/makseq/skflow into makseq-datafeeder_sampling

ilblackdragon committed:
Merge branch 'makseq-datafeeder_sampling'

terrytangyuan committed:
More stable license badge

terrytangyuan committed:
+ Bidirectional rnn option

terrytangyuan committed:
Merge branch 'master' of https://github.com/tensorflow/skflow

terrytangyuan committed:
* Better early stopping params for test

terrytangyuan committed:
* temporarily comment out assertion due to Travis random failure

terrytangyuan committed:
+ Multilayer RNN

terrytangyuan committed:
+ Multilayer bidirection RNN

lopuhin committed:
fix dropout management in TensorFlowEstimator._predict - get correct list of dropout variables, pass keep_prob = 1.0

ilblackdragon committed:
Merge pull request #89 from lopuhin/fix-dropout

Fix dropout management in TensorFlowEstimator._predict

terrytangyuan committed:
* Fixed typo

terrytangyuan committed:
+ sequence_length for RNN model

terrytangyuan committed:
* expose more parameter options in rnn example

terrytangyuan committed:
* temporarily ignore models.py coverage until finish tests for rnn

ilblackdragon committed:
Adding link to Part 3 tutorial.

ilblackdragon committed:
removing disabled package libatlas3gf-base

terrytangyuan committed:
+ initial_state and modified sequence_length

terrytangyuan committed:
* Working unit test for RNN classifier

terrytangyuan committed:
+ rnn regression test

ilblackdragon committed:
Ref #57: Moving to use class weight on logits before cross entropy. The math should work as -weight[class]*x[class] + log( sum ( exp weighted x))

terrytangyuan committed:
Initial code for resnet example

terrytangyuan committed:
* Refactored using skflow helper models

terrytangyuan committed:
* Refactored batch_normalize

terrytangyuan committed:
+ batch_norm_ops and option for batch_norm in conv_ops

terrytangyuan committed:
* Fixed imports and pylint

terrytangyuan committed:
* Replaced with skflowow.ops.conv2d

terrytangyuan committed:
+ Update ops during training

terrytangyuan committed:
+ resnet to README and ignored coverage for batch_norm

terrytangyuan committed:
+ resnet conv_upscale layer

ilblackdragon committed:
CategoricalVocabulary returns len of mapping vocabulary including unknown token

ilblackdragon committed:
Merge branch 'master' of github.com:google/skflow

dansbecker committed:
Remove call to set seed that is then reset when used

terrytangyuan committed:
Merge pull request #91 from dansbecker/remove-redundant-seed

Remove call to set seed. Seed is set again in line that would use it

terrytangyuan committed:
* explicit bias for resnet

terrytangyuan committed:
+ Separate batch_norm for train and test

terrytangyuan committed:
+ batch_norm_ops more params and available for non-convnet

ilblackdragon committed:
Removing empty multiclass test as the case is tested by rnn test

terrytangyuan committed:
* test for other types of rnn

terrytangyuan committed:
+ weights and biases in test to boost coverage

terrytangyuan committed:
+ comments for resnet

terrytangyuan committed:
* table of contents for examples

terrytangyuan committed:
- deleted accidentally included files

ilblackdragon committed:
Adding a batch_size to predict method to split into smaller mini-batches if required

ilblackdragon committed:
Adding support to split numpy arrays into batches for prediction

ilblackdragon committed:
Updating resnet to save logdir, graph and checkpoints and run batches in prediction.

ilblackdragon committed:
Added continue_training=True to the estimator in resnet to make sure it keeps training

ilblackdragon committed:
added xrange to seq2seq_ops

ilblackdragon committed:
Merge branch 'master' into neural-translation

ilblackdragon committed:
fix linters

ilblackdragon committed:
Merge branch 'resnet' into neural-translation

ilblackdragon committed:
Adding neural translation with words and language modeling on characters.

ilblackdragon committed:
Using tf.nn.rnn* instead of direct import

ilblackdragon committed:
Switch to enumerate in seq2seq ops

ilblackdragon committed:
Added small test for bidirectional rnn and forked code from current github version of TF to make it work. Will cleanup when 0.7 is out

ilblackdragon committed:
Merge branch 'master' of github.com:google/skflow

ilblackdragon committed:
Merge branch 'resnet'

ilblackdragon committed:
Merge branch 'master' into neural-translation

ilblackdragon committed:
Fix error in seq2seq ops after changing to enumerate

ilblackdragon committed:
Removing need to import saver_pb2 inside restore function

terrytangyuan committed:
+ Comments for mnist example

ilblackdragon committed:
Merge pull request #74 from tensorflow/neural-translation

Continue working on neural translation model #9. Added language model example #64

ilblackdragon committed:
Add usage of batch norm in conv test and fix usage of is_training collection

ilblackdragon committed:
Changed from using general tmp folder to per-test tmp folder.

terrytangyuan committed:
Update README.md

terrytangyuan committed:
- Unnecessary random call fixed issue #92

ilblackdragon committed:
Fixes #100: do not train keep probability in dropout op

ilblackdragon committed:
Fixes #57: Adding class weight support on estimator level. Now just pass an array of n_classes with weights and it will adjust logits and loss value for this

ilblackdragon committed:
adding num_cores parameter to all the estimators

ilblackdragon committed:
Update version to 0.1

ilblackdragon committed:
fix linter

terrytangyuan committed:
Additional files and descriptions before PyPI submissino

terrytangyuan committed:
Correct converted README.rst file

terrytangyuan committed:
minor change before pypi submission

terrytangyuan committed:
Correct authors information

terrytangyuan committed:
Added PyPI badge in README

terrytangyuan committed:
Update installation instruction

terrytangyuan committed:
Fixed formatting in README.rst

terrytangyuan committed:
Fixed formatting and updated some content

terrytangyuan committed:
+ Changes log

terrytangyuan committed:
bidirection to bidirectional

dansbecker committed:
Add early stopping and reporting based on validation data

anprox committed:
fixed typo

terrytangyuan committed:
Merge pull request #105 from anprox/fix-typo

Fixed typo

terrytangyuan committed:
+ config addon

dansbecker committed:
Minor refactor of monitoring classes. Add monitor arg to estimators. Related fix-ups

terrytangyuan committed:
+ config addon added to estimators

terrytangyuan committed:
* fix in restore/saver

terrytangyuan committed:
* lint fixes

terrytangyuan committed:
+ allow reconfigure config_addon for restored model

terrytangyuan committed:
+ example of custom GPU configurations

terrytangyuan committed:
* allow reconfigurable_params to be reconfigurable

terrytangyuan committed:
* Fixed broken dask io test

terrytangyuan committed:
Update CHANGES.md

terrytangyuan committed:
* restore the correct ConfigAddon

dansbecker committed:
Move monitor argument from model specification to fit

dansbecker committed:
Update early stopping test for consistency with new api

gramhagen committed:
fix path to allow for restoring saved models

ilblackdragon committed:
Merge pull request #108 from gramhagen/restore_fix

fix path to allow for restoring saved models

dansbecker committed:
Update early stopping example

dvbuntu committed:
Add example accessing of weights

It wasn't clear how to access weights using
`classifier.get_tensor_value('foo')` syntax.  This adds some
examples for the CNN model.  They were figured out by
logging the training as though for using TensorBoard, and then
running `strings` on the logfile to look for the right namespace.

liyongsea committed:
Fixed learning_rate in trainer

terrytangyuan committed:
Merge pull request #109 from liyongsea/fix_learning_rate_name_in_trainer

Fixed learning_rate in trainer

dansbecker committed:
Add monitor.BaseMonitor and arg pass through from val monitor

dansbecker committed:
Fix lint

ilblackdragon committed:
Merge pull request #104 from dansbecker/validation-early-stopping

Fixes #85 Add early stopping and reporting based on validation data

ilblackdragon committed:
Fixes #110: Corrected neural_translation_word example to run and train a translation model.

elqursh committed:
Fix predict data feeder to return non-overlapping batches

setup_predict_data_feeder when give an iterator returns
incorrect batches. The incorrect batches take the following
form.

[0, ..., batch_size]
[0, ..., batch_size, batch_size+1]
...

This commit fixes this to return non-overlapping batches of
sizes equal to batch_size.

elqursh committed:
Make estimator.predict more memory efficient

For multi-class classification estimator.predict currently
appends all the predictions and then argmax the resulting
tensor. For a large number of classes this tensor is huge
and may not fit in memory.

This CL changes the behavior to argmax before appending
the results for the batch.

elqursh committed:
Fix iter.next compatiblity with python3

ilblackdragon committed:
Merge pull request #115 from elqursh/optimize-predict

Make estimator.predict memory efficient for large n_classes

terrytangyuan committed:
Merge pull request #114 from elqursh/fix-data-feeder

Fix predict data feeder to return non-overlapping batches

terrytangyuan committed:
Merge pull request #111 from dvbuntu/custom_weights_example

Add example accessing of weights

terrytangyuan committed:
Typo fix and add example to TOC

terrytangyuan committed:
Added recent changes

dgboy2000 committed:
Fix examples given new tensorflow rnn interface. Update bidirectional rnn to return final state.

dgboy2000 committed:
fix spacing

dgboy2000 committed:
update travis ci version of tensorflow

ilblackdragon committed:
Added some robustness to interrupting jobs while save is been run (before would just leave with an empty file, now there is a backup)

ilblackdragon committed:
Merge branch 'master' of github.com:google/skflow

ilblackdragon committed:
Fixes #118: Trim in CategoricalVocabulary remaps ids to 1..n to make sure that max number can be used in categorical variables.

ilblackdragon committed:
Merge pull request #119 from dgboy2000/master

Fix examples given new tensorflow rnn interface. Update bidirectional…

ilblackdragon committed:
Up the required version of Tensorflow - due to recent RNN changes, skflow will now only support 0.7+

ilblackdragon committed:
Ref #118: Additional fixes to neural_translation_word example.

ilblackdragon committed:
Fixing changes in tests due to reordering after trimming

ilblackdragon committed:
Make vocabulary sorting by alphabet first to make it a persistent sort between python versions for the same frequency.

ilblackdragon committed:
Fix mixing type sort in categorical vocabulray

Commit:
fix of the rnn_seq2seq in order to be compatible with tensorflow 0.7

terrytangyuan committed:
Merge pull request #123 from ziky90/master

fix of the rnn_seq2seq in order to be compatible with tensorflow 0.7

terrytangyuan committed:
Added community section

terrytangyuan committed:
Add TF and sklearn installation link, removed multiGPU example from README

mrry committed:
Use the public TensorFlow API to access SaverDef

The current code accesses SaverDef using an obsolete private API. This change makes skflow compatible with TensorFlow 0.7.0 and up.

terrytangyuan committed:
Merge pull request #127 from tensorflow/mrry-patch-1

Use the public TensorFlow API to access SaverDef

terrytangyuan committed:
Fixes #120: version check for TF and sklearn

terrytangyuan committed:
pylint fixes

terrytangyuan committed:
Actually fixed #120

terrytangyuan committed:
Minor changes to version checks

ilblackdragon committed:
Adding current version print to version guard in __init__

ilblackdragon committed:
Added scripts to generate documentation

ilblackdragon committed:
Adding initial documentation

ilblackdragon committed:
updated rst to md

ilblackdragon committed:
added how-to for documentation generation

dustindorroh committed:
Removed redundant instance of StandardScaler

Changed Pipeline to use the existing instance of StandardScaler instead of creating a new instance when Pipeline is created.

terrytangyuan committed:
Merge pull request #132 from dustindorroh/patch-1

Removed redundant instance of StandardScaler

terrytangyuan committed:
Fixes #131: set minimum loss to be last_loss_seen

ilblackdragon committed:
Changes to support tensorflow.github.io/skflow website

ilblackdragon committed:
Merge branch 'master' of github.com:google/skflow

ilblackdragon committed:
In regular DataFeeder for classification convert y into int64 to not raise numpy warnings on itemset

ilblackdragon committed:
Fix #130: Added explicit save/restore to VocabularyProcessor

ilblackdragon committed:
Adding skflow to tensorflow/contrib

ilblackdragon committed:
Merge branch 'skflow-master'

ilblackdragon committed:
Added skflow to tf.contrib

ilblackdragon committed:
Delete old . files

ilblackdragon commented:
Hi @terrytangyuan @makseq @dansbecker @elqursh @dgboy2000 @cbonnett @ivallesp @frol @liyongsea,

We decided to move skflow into tensorflow - for better convergence of APIs and generally better support.
Can you please comment in this thread with "I signed it! message for CLA to propagate for you to tensorflow project (this will make you contributor of tensorflow and keep your stats).

Thanks!

terrytangyuan commented:
I signed it!

dansbecker commented:
I signed it!

ivallesp commented:
I signed it!

makseq commented:
I signed it!

elqursh commented:
I signed it!

On Wed, Mar 9, 2016, 6:42 PM Illia Polosukhin <notifications@github.com>
wrote:

> Hi @terrytangyuan <https://github.com/terrytangyuan> @makseq
> <https://github.com/makseq> @dansbecker <https://github.com/dansbecker>
> @elqursh <https://github.com/elqursh> @dgboy2000
> <https://github.com/dgboy2000> @cbonnett <https://github.com/cbonnett>
> @ivallesp <https://github.com/ivallesp> @frol <https://github.com/frol>
> @liyongsea <https://github.com/liyongsea>,
>
> We decided to move skflow into tensorflow - for better convergence of APIs
> and generally better support.
> Can you please comment in this thread with "I signed it! message for CLA
> to propagate for you to tensorflow project (this will make you contributor
> of tensorflow and keep your stats).
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/1445#issuecomment-194569028>
> .
>

dgboy2000 commented:
I signed it!

On Wed, Mar 9, 2016 at 4:43 PM, elqursh <notifications@github.com> wrote:

> I signed it!
>
> On Wed, Mar 9, 2016, 6:42 PM Illia Polosukhin <notifications@github.com>
> wrote:
>
> > Hi @terrytangyuan <https://github.com/terrytangyuan> @makseq
> > <https://github.com/makseq> @dansbecker <https://github.com/dansbecker>
> > @elqursh <https://github.com/elqursh> @dgboy2000
> > <https://github.com/dgboy2000> @cbonnett <https://github.com/cbonnett>
> > @ivallesp <https://github.com/ivallesp> @frol <https://github.com/frol>
> > @liyongsea <https://github.com/liyongsea>,
> >
> > We decided to move skflow into tensorflow - for better convergence of
> APIs
> > and generally better support.
> > Can you please comment in this thread with "I signed it! message for CLA
> > to propagate for you to tensorflow project (this will make you
> contributor
> > of tensorflow and keep your stats).
> >
> > Thanks!
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/tensorflow/tensorflow/pull/1445#issuecomment-194569028>
> > .
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/1445#issuecomment-194590287>
> .
>

frol commented:
I signed it!

cbonnett commented:
I signed it!

ilblackdragon committed:
Updated imports in skflow and examples, added BUILD

ilblackdragon committed:
Updated readme for skflow examples to explain more about it and installation requirements

ilblackdragon commented:
Jenkins, test this please

ilblackdragon committed:
skflow: Added tests to BUILD and updated tests to import skflow via tensorflow

ilblackdragon commented:
Jenkins, test this please

ilblackdragon committed:
skflow: Added scipy and sklearn package installation to CI

ilblackdragon commented:
Jenkins, test this please

ilblackdragon committed:
Replace pip install scipy with apt-get install python-scipy

ilblackdragon committed:
Added guards around pandas requiring tests to skip them

ilblackdragon commented:
Jenkins, test this please

caisq committed:
pip test-on-install changes for skflow

1) Expanding the search scope for Python unit tests in order to include
the new tests in python/skflow/tests
2) By default, installing scipy and sklearn in virtualenv for the
aforementioned Python unit tests

caisq commented on tensorflow/tools/ci_build/install/install_pip.sh:
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
No need for this file once https://github.com/tensorflow/tensorflow/pull/1472 is in.

caisq commented on tensorflow/tools/ci_build/Dockerfile.cpu:
```diff
@@ -8,6 +8,7 @@ RUN /install/install_bootstrap_deb_packages.sh
 RUN add-apt-repository -y ppa:openjdk-r/ppa
 RUN /install/install_deb_packages.sh
 RUN /install/install_bazel.sh
+RUN /install/install_pip.sh
```
No need for this line once https://github.com/tensorflow/tensorflow/pull/1472 is in.

caisq replied:
1) See comment below. No need to put this in a separate file. Just do pip install here. 
2) The same change needs to be made in Dockerfile.gpu in the same directory.

ilblackdragon committed:
Merge remote-tracking branch 'caisq/help-1445' into skflow

ilblackdragon committed:
Per @caisq comments moved pip install into Docker files

ilblackdragon committed:
Fix defaults for early_stopping and ignoring it when it's none in the BaseMonitor

ilblackdragon committed:
Fix dockerfile syntax

caisq commented:
Sorry about the change in suggestions, but PR 1472, which is already merged after discussion, ended up including all the deb and pip packages required for the builds and tests here. As such, please remove all changes to the following files: 
tensorflow/tools/ci_build/Dockerfile.cpu
tensorflow/tools/ci_build/Dockerfile.gpu
tensorflow/tools/ci_build/install/install_deb_packages.sh
tensorflow/tools/ci_build/builds/test_installation.sh
tensorflow/tools/ci_build/builds/pip.sh
Thanks.

ilblackdragon committed:
Merge remote-tracking branch 'origin/master' into skflow

ilblackdragon committed:
Unroll partial merge from #1472

ilblackdragon committed:
Merge remote-tracking branch 'origin/master' into skflow

ilblackdragon committed:
Added skflow to contrib_py build rule. Removed cyclical dependency

ilblackdragon committed:
Changed to absolute paths in __init__ for skflow

ilblackdragon committed:
Update to absoulte imports in skflow

ilblackdragon committed:
Adding more from __future__ imports and fix aboslutes in monitors

ilblackdragon committed:
add backoff for cPickle to pickle for py3

ilblackdragon committed:
Fix error in py3 for writing bytes in to file with 'w'

ilblackdragon committed:
Fix for reading byte for py3

ilblackdragon committed:
Merge branch 'master' into skflow

ilblackdragon committed:
Removing images from documentation for now as per hosting requirements on tensorflow.org.

ilblackdragon committed:
Merge branch 'skflow' of github.com:ilblackdragon/tensorflow into skflow

terrytangyuan commented:
@ilblackdragon How about the `"cla: no"` thing? Can we still merge it which actually makes sense since they signed `cla` from skflow already?

martinwicke commented:
Everybody has signed the Google CLA when contributing to skflow.
